### PR TITLE
cpu/samd21: cleaned up pad selection macros

### PR DESCRIPTION
**Pull request by haukepetersen**
- gave pad selection macros some more consistent naming scheme
- adapted spi and uart implementation
- adapted board configuration for `samr21-xpro` and `sodaq-autonomo`

### CODE REVIEW

File: boards/samr21-xpro/include/periph_conf.h [hunk 1]
```diff
@@ -104,9 +104,23 @@ extern "C" {
  * @{
  */
 static const uart_conf_t uart_config[] = {
-    /* device, RX pin, TX pin, mux */
-    {&SERCOM0->USART, GPIO_PIN(PA,5),  GPIO_PIN(PA,4),  GPIO_MUX_D, SERCOM_RX_PAD_1, UART_TX_PAD_0},
-    {&SERCOM5->USART, GPIO_PIN(PA,23), GPIO_PIN(PA,22), GPIO_MUX_D, SERCOM_RX_PAD_1, UART_TX_PAD_0},
+    /* device, RX pin, TX pin, mux, RX pad, TX pad */
+    {
+        &SERCOM0->USART,
```
Comment (kaspar030) [inline boards/samr21-xpro/include/periph_conf.h]: how about ```.mosi=GPIO_PIN(PA,5),``` etc?

**Comment by haukepetersen** [comment]
done

File: boards/samr21-xpro/include/periph_conf.h [hunk 2]
```diff
@@ -104,9 +104,23 @@ extern "C" {
...
+        .tx_pad = UART_PAD_TX_0
+    },
+    {
+        .dev    = &SERCOM5->USART,
+        .rx_pin = GPIO_PIN(PA,23),
+        .tx_pin = GPIO_PIN(PA,22),
+        .mux    = GPIO_MUX_D,
+        .rx_pad = UART_PAD_RX_1,
+        .tx_pad = UART_PAD_TX_0
+    }
```
Comment (PeterKietzmann) [inline boards/samr21-xpro/include/periph_conf.h]: :+1:

**Comment by PeterKietzmann** [comment]
@haukepetersen I assume you wanted to squash the last two commits?!

**Comment by haukepetersen** [comment]
done

**Comment by PeterKietzmann** [comment]
It seems the lwip repo is down again so Murdock fails (unrelated to this PR)

**Comment by miri64** [comment]
Shouldn't there be a cache for that -.-

**Comment by PeterKietzmann** [comment]
Then how do you interpret Murdocks error log?

**Comment by haukepetersen** [comment]
IMHO it seems pretty unrelated, but I am not sure...

**Comment by PeterKietzmann** [comment]
Well, the current output doesn't look unrelated anymore. Anything fails for "arduino-zero" with MCU "samd21"

**Comment by haukepetersen** [comment]
That is because the Arduino-zero was merged after I last rebased...

**Comment by haukepetersen** [comment]
rebased and adapted `arduino-zero`

**Comment by PeterKietzmann** [comment]
Ah, wasn't aware of that. ACK and go

**Comment by keestux** [comment]
Is an acceptable series of commits? I would expect these to have been squashed before merged.

**Comment by kYc0o** [comment]
It is acceptable since each commit is affecting a different board and cpu, this way if something is wrong we can know exactly which platform or cpu is involved. We do the same for each part of RIOT (boards, cpu, sys, pkg, etc...), and personally I think it's a good organisation approach.

**Comment by keestux** [comment]
Well, I don't agree. None of the intermediate commits are usable. Bi-sect becomes impossible.
Personally I think that each commit you must be able to build. Each commit must be testable.

**Comment by miri64** [comment]
@keestux that's what `git bisect skip` is for.

**Comment by OlegHahm** [comment]
I basically agree with @keestux here. I had this discussion with @kaspar030 some time ago. However, this discussion should not happen here, but in a dedicated thread on the mailing list.

**Pull request merged**